itest/rust/src/object_test.rs

```diff
@@ -856,3 +862,69 @@ fn double_use_reference() {
     double_use.free();
     emitter.free();
 }
+
+#[derive(GodotClass)]
+#[class(init, base=Object)]
+struct SignalReceiver {
+    used: [Cell<bool>; 3],
+    #[base]
+    base: Base<Object>,
+}
+
+#[godot_api]
+impl SignalReceiver {
+    #[func]
+    fn receive_0_arg(&self) {
+        self.used[0].set(true);
+    }
+    #[func]
+    fn receive_1_arg(&self, arg1: i64) {
+        self.used[1].set(true);
+        assert_eq!(arg1, 987);
+    }
+    #[func]
+    fn receive_2_arg(&self, arg1: Gd<Object>, arg2: GodotString) {
+        self.used[2].set(true);
+        assert_eq!(self.base.share(), arg1);
+        assert_eq!(SIGNAL_ARG_STRING, arg2.to_string());
```

Comment:
Would be nice to check the preconditions first and separate them with an empty line from the logic.
Same above.
```suggestion
        assert_eq!(self.base.share(), arg1);
        assert_eq!(SIGNAL_ARG_STRING, arg2.to_string());
        
        self.used[2].set(true);
```